lib/berlim/accounts/admin.ex

```diff
@@ -17,9 +19,10 @@ defmodule Berlim.Accounts.Admin do
   @doc false
   def changeset(admin, attrs) do
     admin
-    |> cast(attrs, [:email, :password, :name, :active, :phone])
-    |> validate_required([:email, :password, :name, :active])
+    |> cast(attrs, [:email, :encrypted_password, :name, :active, :phone])
+    |> validate_required([:email, :encrypted_password, :name, :active])
     |> unique_constraint(:email)
     |> validate_format(:email, ~r/@/)
+    |> update_change(:encrypted_password, &Bcrypt.hashpwsalt/1)
```

Comment:
a documentação pede assim agora? na época que eu implementei a gente precisou criar um virtual field que trás o campo do frontend, e da um put_change no encrypted_password apenas se o changeset for válido, ou seja, se a senha e todos os outros campos são válidos, como obrigatoriedade, tamanho mínimo e máximo, caracteres especiais permitidos ou não, etc

Reply:
Verdade, achei que só iria encriptar caso o Changeset fosse válido, na documentação do `Ecto.Changeset` ele utiliza um exemplo com o `unique_constraint`, que só ira verificar, caso passe na validação:

> In this case, we haven’t checked the unique constraint in the e-mail field because the data did not validate.

Achei que se aplicava para outros métodos também, mas testei e mesmo não estando válido ele encripta a senha.
Corrigido!